### PR TITLE
feat: Initial libSQL support

### LINKED ISSUE #10451: Support for libSQL/Turso
### Feature Description

Possibility to use libSQL or Turso as a database

### The Solution

Build support for libSQL extending the functionality of sqlite

### Considered Alternatives

None other than using another database type

### Additional Context

_No response_

### Relevant Database Driver(s)

- [ ] aurora-mysql
- [ ] aurora-postgres
- [ ] better-sqlite3
- [ ] cockroachdb
- [ ] cordova
- [ ] expo
- [ ] mongodb
- [ ] mysql
- [ ] nativescript
- [ ] oracle
- [ ] postgres
- [ ] react-native
- [ ] sap
- [ ] spanner
- [ ] sqlite
- [ ] sqlite-abstract
- [ ] sqljs
- [ ] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

No, I don’t have the time, but I can support (using donations) development.

### PR DESCRIPTION
**Pull request by netroy**
### Description of change

This Fixes #10451 by introducing libSQL support for [local files](https://github.com/tursodatabase/libsql-client-ts#local-sqlite-files). 

### Pull-Request Checklist

- [x] Code is up-to-date with the `master` branch
- [x] `npm run format` to apply prettier formatting
- [x] `npm run test` passes with this change
- [x] This pull request links relevant issues as `Fixes #10451`
- [ ] There are new or updated unit tests validating the change
- [ ] Documentation has been updated to reflect this change
- [x] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/typeorm/typeorm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: src/query-builder/relation-id/RelationIdLoader.ts [hunk 1]
```diff
@@ -45,18 +45,18 @@ export class RelationIdLoader {
...
-                                                DriverUtils.buildAlias(
-                                                    this.connection.driver,
-                                                    undefined,
-                                                    relationIdAttr.parentAlias,
-                                                    joinColumn.databaseName,
-                                                )
-                                            ],
-                                            joinColumn.referencedColumn!,
-                                        )
+                                    const columnAlias = DriverUtils.buildAlias(
```
Comment (netroy) [inline src/query-builder/relation-id/RelationIdLoader.ts:48]: This entire file can be reverted if https://github.com/tursodatabase/libsql-client-ts/pull/166 gets merged.

File: src/driver/libsql/LibSqlDriver.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    ) {
+        if (typeof insertResult === "bigint")
+            insertResult = Number(insertResult)
+        return super.createGeneratedMap(
+            metadata,
+            insertResult,
+            entityIndex,
+            entityNum,
+        )
+    }
```
Comment (netroy) [inline src/driver/libsql/LibSqlDriver.ts:66]: This can be removed if https://github.com/tursodatabase/libsql-client-ts/issues/165 is addressed.

**Comment by netroy** [comment]
closing this, since there doesn't seem to be any interest.

**Comment by penberg** [comment]
We do have people using Turso/libSQL asking about TypeORM support. So we'd be excited to see this happen in TypeORM.

**Comment by hieuht-htv** [comment]
+1

**Comment by ibilux** [comment]
@netroy this should be merged !

**Pull request closed without merging**